src/main/java/com/aws/greengrass/device/ClientDevicesAuthService.java

```diff
@@ -215,4 +230,18 @@ private void uploadCoreDeviceCAs(List<String> certificatePemList) {
                     String.format("Failed to put core %s CA certificates to cloud", thingName), e);
         }
     }
+
+    @Override
+    public void postInject() {
+        super.postInject();
+        try {
+            authorizationHandler.registerComponent(this.getName(),
+                    new HashSet<>(Collections.singletonList(SUBSCRIBE_TO_CERTIFICATE_UPDATES)));
+        } catch (AuthorizationException e) {
+            logger.atError("initialize-cda-service-authorization-error", e)
+                    .log("Failed to initialize the client device auth service with the Authorization module.");
+        }
+        greengrassCoreIPCService.setSubscribeToCertificateUpdatesHandler(context ->
```

Comment:
We are proceeding even if registration fails?

Reply:
Oops. Good catch.

Reply:
It can't ever fail, it only fails if the inputs are wrong and the inputs don't ever change; we know for certain that it will succeed

Reply:
Let's still structure the code in the right way.

Reply:
Even if we can't register the auth handler, there's no reason why you can't register the IPC handler.

Also look at other examples in Nucleus, they do the same.

Reply:
Looks like exceptions can't be thrown from post inject unless we make a change to the nucleus. I checked other plugins with IPC and they do the same.